internal/services/storage/storage_account_data_source.go

```diff
@@ -675,18 +675,14 @@ func dataSourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) e
 		secondaryEndpoints = model.Properties.SecondaryEndpoints
 	}
 	endpoints := flattenAccountEndpoints(primaryEndpoints, secondaryEndpoints, routingPreference)
-	if err := endpoints.set(d); err != nil {
-		return err
-	}
+	endpoints.set(d)
```

Comment:
We should also fix the `endpoints.set(d)` to not return an error if it's just returning nil